### ISSUE TITLE
Empty pages built when paginating collection on homepage

### ISSUE THREAD

**Issue by hellocosmin** [gridsome/gridsome#218]
## Description

Following the [docs](https://gridsome.org/docs/pagination), I'm paginating a blog posts collection, which I then show on my site's homepage.

The actual file generated at `dist/2/index.html` (and so on), is empty: it contains the 404 page contents, instead of the HTML with blog posts, like `dist/index.html` does.

### Steps to reproduce

For a quick test, see [this demo page](https://wonderful-liskov-6ff407.netlify.com/2/) - you should see a 404. Of course, when going to the [homepage](https://wonderful-liskov-6ff407.netlify.com) and then navigating (with JS enabled) to it from the bottom nav, it works, because Vue takes care of that...

Oddly, paginated taxonomy pages, i.e. [category-name/2/](https://wonderful-liskov-6ff407.netlify.com/some-category/2/) are built perfectly fine.

I have the following config:

```js
// gridsome.config.js
module.exports = {
  siteName: 'Gridsome Starter Blog',
  titleTemplate: `%s | Gridsome`,
  plugins: [
    {
      use: '@gridsome/source-filesystem',
      options: {
        path: 'content/posts/**/*.md',
        typeName: 'Post',
        route: '/:category/:title-:year:month:day',
        refs: {
          category: {
            typeName: 'Category',
            route: '/:title',
            create: true
          },
        }
      }
    },
  ],
}
```

In `src/pages/Index.vue`, I have the following `<page-query>`:

```
<page-query>
  query Home ($page: Int) {
    allPost (page: $page, perPage: 2) @paginate {
      totalCount
      pageInfo {
        totalPages
        currentPage
      }
      edges {
        node {
          _id
          title
          date (format: "D MMMM, YYYY")
          datetime: date (format: "YYYY-MM-DD HH:mm:ss")
          content
          excerpt
          path
          image
          category {
            title
          }
        }
      }
    }
  }
</page-query>
```

Finally, I have 3 markdown posts in `content/posts`, to make sure a second index page is generated.

### Expected result

`dist/2/index.html` should contain the posts from page 2 of the site, as displayed when loaded by Vue.

### Actual result

`dist/2/index.html` has the 404 content. File is indeed output, but the content is not there. This happens to all following pages, so `/3/`, `/4/` and so on are also 'empty'.

This is what it contains:

```
<!DOCTYPE html>
<html data-html-server-rendered="true" lang="ro" data-vue-tag="lang">
  <head>
    <title data-vue-tag="true">Gridsome Starter Blog | Gridsome</title><meta data-vue-tag="true" charset="utf-8"/><meta data-vue-tag="true" name="generator" content="Gridsome v0.5.0"/><meta data-vue-tag="true" data-key="viewport" name="viewport" content="width=device-width, initial-scale=1, viewport-fit=cover"/><meta data-vue-tag="true" data-key="description" name="description" content=""/><meta data-vue-tag="true" data-key="format-detection" name="format-detection" content="telephone=no"/><link data-vue-tag="true" rel="icon" type="image/png" sizes="16x16" href="/assets/static/favicon.ce0531f.efcb514.png"/><link data-vue-tag="true" rel="icon" type="image/png" sizes="32x32" href="/assets/static/favicon.ac8d93a.efcb514.png"/><link data-vue-tag="true" rel="icon" type="image/png" sizes="96x96" href="/assets/static/favicon.b9532cc.efcb514.png"/><link data-vue-tag="true" rel="apple-touch-icon" type="image/png" sizes="76x76" href="/assets/static/favicon.f22e9f3.efcb514.png"/><link data-vue-tag="true" rel="apple-touch-icon" type="image/png" sizes="152x152" href="/assets/static/favicon.62d22cb.efcb514.png"/><link data-vue-tag="true" rel="apple-touch-icon" type="image/png" sizes="120x120" href="/assets/static/favicon.1539b60.efcb514.png"/><link data-vue-tag="true" rel="apple-touch-icon" type="image/png" sizes="167x167" href="/assets/static/favicon.dc0cdc5.efcb514.png"/><link data-vue-tag="true" rel="apple-touch-icon" type="image/png" sizes="180x180" href="/assets/static/favicon.7b22250.efcb514.png"/><link data-vue-tag="true" rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,700%7CHind+Madurai:400,500&amp;amp;subset=latin-ext"/><noscript data-vue-tag="true" ><style>.g-image--loading{display:none;}</style></noscript><link rel="preload" href="/assets/css/styles.76a9a8cc.css" as="style"><link rel="preload" href="/assets/js/app.e7cec1f7.js" as="script"><link rel="preload" href="/assets/js/component--404.87245fb2.js" as="script"><link rel="prefetch" href="/assets/js/component--category.59116350.js"><link rel="prefetch" href="/assets/js/component--home.006d5547.js"><link rel="prefetch" href="/assets/js/component--post.91f8a3a2.js"><link rel="prefetch" href="/assets/js/page-query.584d3c2f.js"><link rel="stylesheet" href="/assets/css/styles.76a9a8cc.css">
  </head>
  <body class="antialiased font-hind-madurai" data-vue-tag="class">
    <h1 data-server-rendered="true" id="app">404 - not found</h1>
    <script src="/assets/js/app.e7cec1f7.js" defer></script><script src="/assets/js/component--404.87245fb2.js" defer></script>
  </body>
</html>
```

**Comment by hellocosmin** [comment]
I should also add that I've tried other `route`s for the collections. For example, this yields the same results:

```js
plugins: [
    {
      use: '@gridsome/source-filesystem',
      options: {
        path: 'content/posts/**/*.md',
        typeName: 'Post',
        route: '/:title',
        refs: {
          category: {
            typeName: 'Category',
            route: '/categories/:slug',
            create: true
          },
        }
      }
    },
  ],
```

**Comment by hjvedvik** [comment]
Looks like there is a bug with root level pagination. I'll fix it in the next update.

**Comment by hellocosmin** [comment]
Got excited when I saw `0.5.2`, so I went ahead and updated it and `@gridsome/transformer-remark` to `0.2.1`. Ran `gridsome build` and it now seems files are output properly - the page at `/2/` has the right content.

However there's now something else wrong, this time when Vue takes over:

```
Error: Cannot find module './/index.json'
    at page-query.51f1a430.js:1
```

You can check it at the [same link](https://wonderful-liskov-6ff407.netlify.com/). Apologies if you're already aware of it - no pressure :)

**Comment by hjvedvik** [comment]
Oh, I'll try to publish a fix asap :)

**Comment by hjvedvik** [comment]
Published `0.5.3` now. Should fix this issue :)

**Comment by hellocosmin** [comment]
Yes, `0.5.3` has fixed it 🎉

Thank you!
